test/indexer/test_indexer_controller.py

```diff
@@ -0,0 +1,137 @@
+import json
+import logging
+
+import boto3
+import mock
+from moto import (
+    mock_sqs,
+    mock_sts,
+)
+
+from azul import (
+    config,
+    queues,
+)
+from azul.azulclient import (
+    AzulClient,
+)
+from azul.indexer import (
+    BundleFQID,
+)
+from azul.indexer.index_controller import (
+    IndexController,
+)
+from indexer import (
+    ForcedRefreshIndexService,
+    IndexerTestCase,
+)
+
+log = logging.getLogger(__name__)
+
+
+class Record():
+    body = None
+    _event_dict = None
+
+    def __init__(self, body):
+        self.body = body
+        self._event_dict = dict(attributes=dict(ApproximateReceiveCount=1))
+
+    def to_dict(self):
+        return self._event_dict
+
+
+class IndexControllerTestCase(IndexerTestCase):
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.index_service.create_indices(self.catalog)
+        self.client = AzulClient()
+        self.controller = IndexController()
+
+    def tearDown(self):
+        self.index_service.delete_indices(self.catalog)
+        super().tearDown()
+
+    def generate_events(self):
+        pass
+
+    def _create_mock_queues(self):
+        sqs = boto3.resource('sqs')
+        for queue_name in config.all_queue_names:
+            sqs.create_queue(QueueName=queue_name)
+
+    def generate_record(self, **kwargs):
+        return Record(body=json.dumps(kwargs))
+
+    def test_contributions(self):
+        self._test_contribution_action_invalid()
+        self._test_contribution_action_reindex()
+        self._test_contribute_and_aggregate()
+
+    def _test_contribution_action_invalid(self):
+        event = [self.generate_record(action='banana', notification='slug', catalog=self.catalog)]
+        self.assertRaises(AssertionError, self.controller.contribute, event)
+
+    def _test_contribution_action_reindex(self):
+        event = [self.generate_record(action='reindex',
+                                      prefix='ff')]
+        with mock.patch('azul.azulclient.AzulClient.do_remote_reindex') as acr:
+            acr.return_value = True
+            self.controller.contribute(event)
+            self.assertTrue(acr.called)
+
+    @mock_sts
+    @mock_sqs
+    def _test_contribute_and_aggregate(self):
+        """
+        Index two bundles that make contributions to the same project.
+        Inspect that the contributions match the tallies that are returned to
+        SQS. During aggregation only the project entity is deferred due to
+        multiple contributions.
+        """
+        self._create_mock_queues()
+        event = list()
+        bundles = list()
+        list_of_contributions = list()  # rename this to something more meaningful, (ss) for multiple contributions
```

Comment:
```suggestion
        contribution_lists = list()
```
Not sure what `(ss)` is referring to here.

Reply:
eeep thats the secret sauce ill take it out.